test/Control/Monad/ST.purs

```diff
@@ -46,3 +48,12 @@ testModifySTRef' = do
   val <- ref %%= \old -> { newValue: old, returnValue: not old}
   assert $ val == false
   testReadSTRef ref true
+
+testGetRef :: Eff (console :: CONSOLE, assert :: ASSERT) Unit
+testGetRef = unsafeRunRef do
+  log "getRef"
+  ref <- runST do
+    ref' <- newSTRef true
+    getRef ref'
```

Comment:
@paf31 I demoed your conundrum here and resolved it by coercing the phantom type into an effect row.